agent.go

```diff
@@ -97,11 +99,15 @@ func newAgent(serviceName, host string, port int, logger LeveledLogger) *agentS
 
 	logger.Debug("initializing agent")
 
+	// maxClientTimeout is max timeout that can this client has, for each request it can be reduced using `context.WithTimeout`
+	maxClientTimeout := 60 * time.Second
 	agent := &agentS{
-		from:   &fromS{},
-		host:   host,
-		port:   strconv.Itoa(port),
-		client: &http.Client{Timeout: 5 * time.Second},
+		from:            &fromS{},
+		host:            host,
+		port:            strconv.Itoa(port),
+		clientTimeout:   5 * time.Second,
+		announceTimeout: 15 * time.Second,
+		client:          &http.Client{Timeout: maxClientTimeout},
```

Comment:
Apologies, I've missed this change during the last review. What's the reason behind it and where did 60s come from?

Reply:
The single reason is always to have at least some timeout. Because I believe by default there is no timeout at all.

Reply:
I see, shall we use 15s then and shorten it to 5s where appropriate? Wouldn't this simplify the logic?

Reply:
I changed this now.